sessionpeermanager/sessionpeermanager_test.go

```diff
@@ -39,20 +40,27 @@ func (fpn *fakePeerNetwork) FindProvidersAsync(ctx context.Context, c cid.Cid, n
 
 type fakeConnManager struct {
 	taggedPeers []peer.ID
+	wait        sync.WaitGroup
 }
 
 func (fcm *fakeConnManager) TagPeer(p peer.ID, tag string, n int) {
+	fcm.wait.Add(1)
 	fcm.taggedPeers = append(fcm.taggedPeers, p)
 }
+
 func (fcm *fakeConnManager) UntagPeer(p peer.ID, tag string) {
+	fcm.wait.Done()
```

Comment:
You might want to defer this or put it after the removal from fcm.taggedPeers otherwise you might get a test failure.

Reply:
oh right. got it.